GAE/src/org/waterforpeople/mapping/app/web/BootstrapGeneratorServlet.java

```diff
@@ -158,7 +159,24 @@ private void generateFile(BootstrapGeneratorRequest req) {
         MailUtil.sendMail(PropertyUtil.getProperty(EMAIL_FROM_ADDRESS_KEY),
                 "FLOW", req.getEmail(), EMAIL_SUB, body);
     }
-    
+
+    /**
+     * Uses name to generate a safe string to use as filename, removing any unsafe chars
+     * @param name
+     * @return
+     */
+    private String generateSanitizedFilename(@Nullable String name) {
+        String filename;
+        if (StringUtil.isEmpty(name)) {
+            filename = DEFAULT_SURVEY_FILE_NAME;
+        } else {
+            filename = name.replaceAll(FILENAME_REGEX, FILENAME_REPLACEMENT);
```

Comment:
It may be over thinking this but I got some errors when I tried to make a form name with spaces around the "/" e.g `Form with / back slash` where as `Form with/back slash` worked on the device. So maybe we can add another `replaceAll()` after the first one to replace any patterns like `Form_with___back_slash`?